src/BuildScriptGenerator/DotNetCore/VersionProviders/DotNetCoreSdkStorageVersionProvider.cs

```diff
@@ -58,24 +59,43 @@ public void GetVersionInfo()
                 // keys represent runtime version, values represent sdk version
                 var supportedVersions = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
+                var sdkVersionMetadataName = SdkStorageConstants.SdkVersionMetadataName;
+                var runtimeVersionMetadataName = SdkStorageConstants.DotnetRuntimeVersionMetadataName;
+
+                if (this.commonOptions.DebianFlavor == OsTypes.DebianStretch)
+                {
+                    sdkVersionMetadataName = SdkStorageConstants.LegacySdkVersionMetadataName;
+                    runtimeVersionMetadataName = SdkStorageConstants.LegacyDotnetRuntimeVersionMetadataName;
+                }
+
                 foreach (var metadataElement in xdoc.XPathSelectElements($"//Blobs/Blob/Metadata"))
                 {
                     var childElements = metadataElement.Elements();
+
                     var runtimeVersionElement = childElements.Where(e => string.Equals(
-                            DotNetCoreConstants.StorageSdkMetadataRuntimeVersionElementName,
+                            runtimeVersionMetadataName,
                             e.Name.LocalName,
                             StringComparison.OrdinalIgnoreCase))
                         .FirstOrDefault();
 
                     if (runtimeVersionElement != null)
                     {
                         var sdkVersionElement = childElements.Where(e => string.Equals(
-                                DotNetCoreConstants.StorageSdkMetadataSdkVersionElementName,
+                                sdkVersionMetadataName,
+                                e.Name.LocalName,
+                                StringComparison.OrdinalIgnoreCase))
+                            .FirstOrDefault();
+
+                        var osTypeElement = childElements.Where(e => string.Equals(
+                                SdkStorageConstants.OsTypeMetadataName,
                                 e.Name.LocalName,
                                 StringComparison.OrdinalIgnoreCase))
                             .FirstOrDefault();
 
-                        supportedVersions[runtimeVersionElement.Value] = sdkVersionElement.Value;
+                        if (this.commonOptions.DebianFlavor == OsTypes.DebianStretch || this.commonOptions.DebianFlavor == osTypeElement.Value)
```

Comment:
Just to confirm: in addition to the platform binaries corresponding to the current Debian flavor of the image, we're _always_ going to say that stretch binaries are supported too, correct? It may be worth adding a comment above this if-statement calling out what's happening here just to be 100% clear to anyone who runs across this code in the future.

Reply:
Yes, so the specific case this is addressing is the stretch platform binaries that no longer are uploaded (and therefore aren't overwritten with new metadata). So we can't rely on os type to pull stretch SDKs. Its also a little tricky, since we have to keep in mind that stretch SDKs have a different version metadata than all others (Version instead of Sdk_version).

We determine stretch supported versions by looking for SDKs with the `Version` metadata (only stretch SDKs will have this). we determine all other os type supported versions by looking for `Sdk_version` metadata for all SDKs with a matching `Os_type` metadata.

Reply:
@cormacpayne though you do raise a good point that we aren't checking to make sure that SDK version exists. I added this check, which shouldn't be possible to reach if we upload using our scripts... but adding the check means that this would simply not add a mis-formatted sdk as a supported version, instead of breaking the oryx build.